### ISSUE TITLE
Change shared folder permissions buttons to checkboxes

### ISSUE THREAD

**Issue by levtecaccess** [openmediavault/openmediavault#795]
As a blind user using a screen reader, when setting permissions for shared folders, I am unable to find out the current state of the permissions due to the button design.
Please either adjust the buttons to indicate pressed state or switch to checkboxes to better indicate the current state.
I have looked through the code personally but can't locate the controls to make alterations and run tests, otherwise I would have done so and submitted my code for review.

**Comment by PrplHaz4** [comment]
I'm guessing the Angular UI rewrite will have a much better accessibility story than current ExtJS - and it seems almost all UI updates are being slotted into that rather than current OMV 5.

UI Rewrite: https://github.com/openmediavault/openmediavault/issues/684

Code for ExtJS UI: https://github.com/openmediavault/openmediavault/blob/master/deb/openmediavault/var/www/openmediavault/js/omv/module/admin/privilege/sharedfolder/SharedFolder.js

**Comment by votdev** [comment]
> otherwise I would have done so and submitted my code for review.

Feel free to do so. Looking forward to get an impression how you would improve it. Details can be discussed in the PR then.

**Comment by levtecaccess** [comment]
Hi

Thanks for the comments, I believe I have located what I need in deb/openmediavault/var/www/openmediavault/js/omv/grid/Privileges.js which appears to be the code used to define the permissions grid.
`				xtype: "checkcolumn",
				text: _("Read/Write"),
				sortable: true,
				groupable: false,
				dataIndex: "writeable",
				stateId: "writeable",
				align: "center",
				flex: 1,
				listeners: {
					scope: me,
					checkchange: me.onCheckChange
				}
			},{
				xtype: "checkcolumn",
				text: _("Read-only"),
				sortable: true,
				groupable: false,
				dataIndex: "readonly",
				stateId: "readonly",
				align: "center",
				flex: 1,
				listeners: {
					scope: me,
					checkchange: me.onCheckChange
				}
			},{
				xtype: "checkcolumn",
				text: _("No access"),
				sortable: true,
				groupable: false,
				dataIndex: "deny",
				stateId: "deny",
				align: "center",
				flex: 1,
				listeners: {
					scope: me,
					checkchange: me.onCheckChange
				}

I would change the above to use checkboxes, however I am not familiar with ExtJS and am unsure how to go about this.
I assume I can't just alter the xtype value since I assume the checkbox type would have different parameters to the checkcolumn type, I guess reading the ExtJS documentation regarding checkboxes would help me.
I am dubious to make any changes here as a blind user, since I don't know if what I am attempting will break the UI for sighted people.
I am perfectly happy to make the changes on a live system for testing and am happy to report my findings, but would like some guidance on how best to proceed.
`

**Comment by votdev** [comment]
The [checkcolumn](https://docs.sencha.com/extjs/6.2.0/classic/Ext.grid.column.Check.html) is the official ExtJS way to display checkboxes. If you want to go another route it requires the implementation of a separate column type. I think this is overkill and i will not do this because i will never touch ExtJS code anymore except minor fixes due the fact that it will be replaced with OMV6. I assume the problem here is that the ExtJS implementaion simply does not fully support `ARIA` roles and/or `A11y`. Sadly i'm not familar with this, so any hint is welcome.

**Comment by votdev** [comment]
Maybe setting `useAriaElements` to `true` will fix the issue. The problem is the [docs](https://docs.sencha.com/extjs/6.2.0/classic/Ext.grid.column.Column.html) do not know that property.

**Comment by votdev** [comment]
Will `aria-pressed` fix the issue? https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

**Comment by levtecaccess** [comment]
I think aria-pressed would go some way to helping the screen reader know the state.
I will add the useAriaElements property to the code and test my results.
I will let you know if this helps.

Just a quick question, what turned you off of ExtJS if you don't mind me asking?

**Comment by votdev** [comment]
> Just a quick question, what turned you off of ExtJS if you don't mind me asking?

They simply do not ship a GPL'ed version of the framework anymore. IMO they turned into a cash driven company forgetting their origin roots.

**Comment by levtecaccess** [comment]
Ok, so I just tested my changes and the screen reader announces "undefined" for each checkcolumn.
This is a step forward since the screen reader would simply announce the name of each column without the useAriaElements: true property defined.
I am unsure how to implement aria-pressed, is this another property or implemented via CSS?

**Comment by votdev** [comment]
> I am unsure how to implement aria-pressed, is this another property or implemented via CSS?

That's where it becomes difficult because this must be fixed in the ExtJS code. You have to override or intercept the ExtJS code, maybe the renderer or somewhere else in https://docs.sencha.com/extjs/6.2.0/classic/src/Check.js.html.

Examples:
- https://github.com/openmediavault/openmediavault/blob/master/deb/openmediavault/var/www/openmediavault/js/ext-overrides.js#L791
- https://github.com/openmediavault/openmediavault/blob/master/deb/openmediavault/var/www/openmediavault/js/ext-overrides.js#L827

That's why i really don't want to touch the UI code anymore. But if you're interested in doing that then that would be great.

**Comment by levtecaccess** [comment]
After looking at the documentation, it seems I could experiment with the checkboxAriaRole property.
I will experiment by first adding it directly to the code and then moving it to ext-overrides if I am unsuccessfull.

**Comment by vincejansen** [comment]
Great that this can be fixed! I've opened an issue about making the GUI more accessible way back but I left that issue a sort of alone, because I'm not a programmer, but i could test things out if you want. 

Keep up the amazing work for so far!

**Comment by mi-hol** [comment]
@levtecaccess may I ask for a status update of your attempt to improve the code?

**Comment by mi-hol** [comment]
closing due to lacking feedback
